controller/command/command.go

```diff
@@ -0,0 +1,47 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package command
+
+import (
+	"encoding/binary"
+	"google.golang.org/protobuf/proto"
+)
+
+type Command interface {
+	Apply() error
+	Encode() ([]byte, error)
+}
+
+type Validatable interface {
+	Validate() error
+}
+
+type TypedMessage interface {
+	proto.Message
+	GetCommandType() int32
+}
```

Comment:
Mysterious file. At least documenting the major interfaces would be nice.

Reply:
Documented. Moved TypedMessage and Encode to the cmd_pb package